aspnetcore/performance/rate-limit.md

```diff
@@ -19,6 +19,150 @@ The `Microsoft.AspNetCore.RateLimiting` middleware provides rate limiting middle
+```sh
+   dotnet add package Microsoft.AspNetCore.RateLimiting
+```
+
+### 2. Configure rate limiting services
+In the Program.cs file, configure the rate limiting services by adding the appropriate rate limiting policies. Policies can either be defined as global, for example the following which permits 10 requests per minute:
+
+``` csharp
+builder.Services.AddRateLimiter(options =>
+{
+    options.GlobalLimiter = PartitionedRateLimiter.Create<HttpContext, string>(httpContext =>
+        RateLimitPartition.GetFixedWindowLimiter(
+            partitionKey: httpContext.User.Identity?.Name ?? httpContext.Request.Headers.Host.ToString(),
+            factory: partition => new FixedWindowRateLimiterOptions
+            {
+                AutoReplenishment = true,
+                PermitLimit = 10,
+                QueueLimit = 0,
+                Window = TimeSpan.FromMinutes(1)
+            }));
+});
+```
+
+Or as named polices, which need to be explicitly applied to the pages or endpoints. For example, to add a fixed window limiter:
+
+``` csharp
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddRateLimiter(options =>
+{
+    options.AddFixedWindowLimiter("fixed", opt =>
+    {
+        opt.PermitLimit = 4;
+        opt.Window = TimeSpan.FromSeconds(12);
+        opt.QueueProcessingOrder = QueueProcessingOrder.OldestFirst;
+        opt.QueueLimit = 2;
+    });
+});
+
+var app = builder.Build();
+```
+
+The global limiter applies to all endpoints automatically when it's configured via options.GlobalLimiter, and no endpoint-specific policy is specified.
+
+### 3. Enable rate limiting middleware
+ In the Program.cs file, enable the rate limiting middleware by calling UseRateLimiter:
+
+``` csharp
+app.UseRouting();
+
+app.UseRateLimiter();
+
+app.UseEndpoints(endpoints =>
+{
+    endpoints.MapControllers();
+});
+
+app.Run();
+```
+
+### Apply rate limiting policies to endpoints or pages
+
+#### Apply rate limiting to WebAPI Endpoints
+
+Apply a named policy to the endpoint or group, for example:
+
+``` csharp
+
+app.MapGet("/api/resource", () => "This endpoint is rate limited")
+   .RequireRateLimiting("fixed"); // Apply specific policy to an endpoint
+
+```
+
+#### Apply rate limiting to MVC Controllers
+
+ Apply the configured rate limiting policies to specific endpoints or globally. For example, to apply the "fixed" policy to all controller endpoints:
+
+``` csharp
+app.UseEndpoints(endpoints =>
+{
+    endpoints.MapControllers().RequireRateLimiting("fixed");
+});
+
+```
+
+#### Apply rate limiting to Blazor Server Pages
+
+To set rate limiting to all pages,  `RequireRateLimiting(Policy)` can be specified on the MapRazorComponents call, for example:
+
+``` csharp
+app.MapRazorComponents<App>()
+    .AddInteractiveServerRenderMode()
+    .RequireRateLimiting("MyPagePolicy");
+```
+
+To set policy for individual Blazor Pages, the attribute must be applied to the Page and not the page handlers.
```

Comment:
Not sure what is meant here by "page handlers," and it doesn't seem necessary to even mention it ...

```suggestion
To set a policy for a single Razor component, the [`[EnableRateLimiting]` attribute](xref:Microsoft.AspNetCore.RateLimiting.EnableRateLimitingAttribute) is applied to the component with an optional policy name:
```